pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +361,84 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,TechnicalID to global getTargetInfo
 */
-func getTechnicalID() string {
+func getTargetMapInfo() map[string]string {
 	var target Target
 	ReadTarget(pathTarget, &target)
 
-	gardenClientset, err := target.GardenerClient()
-	checkError(err)
-	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
-	checkError(err)
-	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
-	checkError(err)
-	return shoot.Status.TechnicalID
+	for _, t := range target.Stack() {
+		getTargetInfo[string(t.Kind)] = string(t.Name)
+	}
+
+	// user for case garden-[project]-seed-shoot
+	if isTargetFilsHas("seed", "shoot") {
+		delete(getTargetInfo, "seed")
+
+		shootname := getTargetInfo["shoot"]
+		project, err := getProjectObject()
+		checkError(err)
+		shoot, err := getShootObject(*project.Spec.Namespace, shootname)
+		checkError(err)
+		getTargetInfo["seed"] = *shoot.Spec.SeedName
+		getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+		shoot, err = getShootObject("garden", getTargetInfo["seed"])
+		checkError(err)
+		getTargetInfo["seedTechnicalID"] = shoot.Status.TechnicalID
+	}
+
+	// user for case garden-project-shoot
+	if isTargetFilsHas("project", "shoot") {
+		shootname := getTargetInfo["shoot"]
+		project, err := getProjectObject()
+		checkError(err)
+		shoot, err := getShootObject(*project.Spec.Namespace, shootname)
+		checkError(err)
+		getTargetInfo["seed"] = *shoot.Spec.SeedName
+		getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
```

Comment:
you have the exact same code in the upper if (lines 378-384)

Reply:
fixed